Applications/LogisticRegression/jni/main.cpp

```diff
@@ -161,64 +160,79 @@ int main(int argc, char *argv[]) {
 
   srand(time(NULL));
 
-  auto data_train = ml::train::createDataset(ml::train::DatasetType::GENERATOR,
-                                             getSample_train);
-
   /**
    * @brief     Create NN
    */
-  std::vector<std::vector<float>> inputVector, outputVector;
-  nntrainer::NeuralNetwork NN;
+  std::unique_ptr<ml::train::Model> model;
+  model = ml::train::createModel(ml::train::ModelType::NEURAL_NET);
 
   /**
    * @brief     Initialize NN with configuration file path
    */
 
   try {
-    NN.loadFromConfig(config);
-    NN.compile();
-    NN.initialize();
+    model->load(config, ml::train::ModelFormat::MODEL_FORMAT_INI);
+    model->compile();
+    model->initialize();
   } catch (...) {
     std::cerr << "Error during init" << std::endl;
     return 0;
   }
 
   if (training) {
-    NN.setDataset(ml::train::DatasetModeType::MODE_TRAIN,
-                  std::move(data_train));
+    auto data_train = ml::train::createDataset(
+      ml::train::DatasetType::GENERATOR, getSample_train);
+
+    model->setDataset(ml::train::DatasetModeType::MODE_TRAIN,
+                      std::move(data_train));
 
     try {
-      NN.train({"save_path=" + weight_path});
+      model->train({"save_path=" + weight_path});
     } catch (...) {
       std::cerr << "Error during train" << std::endl;
       return 0;
     }
   } else {
     try {
-      NN.load(weight_path, ml::train::ModelFormat::MODEL_FORMAT_BIN);
+      model->load(weight_path);
     } catch (std::exception &e) {
       std::cerr << "Error during loading weights: " << e.what() << "\n";
       return 1;
     }
     std::ifstream dataFile(data_file);
     int cn = 0;
+
     for (unsigned int j = 0; j < total_val_data_size; ++j) {
-      nntrainer::Tensor d;
-      std::vector<float> o;
-      std::vector<float> l;
-      o.resize(feature_size);
-      l.resize(1);
 
-      getData(dataFile, o.data(), l.data(), j);
+      float input[3] = {0.0, 0.0, 0.0};
+      float label[1] = {0.0};
+
+      if (!getData(dataFile, input, label, j))
+        std::cout << "error dring read file " << std::endl;
 
       try {
-        float answer =
-          NN.forwarding({MAKE_SHARED_TENSOR(nntrainer::Tensor({o}))},
-                        {MAKE_SHARED_TENSOR(nntrainer::Tensor({l}))})[0]
-            ->apply(stepFunction)
-            .getValue(0, 0, 0, 0);
-        std::cout << answer << " : " << l[0] << std::endl;
-        cn += answer == l[0];
+
+        std::vector<float *> in;
+        std::vector<float *> l;
+        std::vector<float *> answer;
+        float a[1] = {0.0};
+
+        in.push_back(input);
+        l.push_back(label);
+
+        answer = model->inference(1, in, l);
+
+        in.clear();
+        l.clear();
```

Comment:
Does it needs to be cleared? It seems that *in* and *l* is not used after this.

Reply:
there is a loop at L207. So they need to be clear.